beacon-chain/blockchain/service.go

```diff
@@ -77,8 +78,16 @@ func (c *ChainService) Start() {
 	c.genesisTime, err = c.beaconDB.GenesisTime()
 	if err != nil {
 		log.Fatalf("Unable to retrieve genesis time - blockchain service could not start: %v", err)
+		c.failStatus = err
 		return
 	}
+
+	if c.web3Service.Status() != nil {
+		log.Fatalf("Unable to connect to web3Service")
+		c.failStatus = c.web3Service.Status()
```

Comment:
You don’t need to set failStatus for something you can query within the status func.

Your Status() should look like 

```go
func (c * ChainService) Status() error {
  if c.web3Service.Status() != nil {
    return c.web3Service.Status()
  } 

  ... other checks here ...
 
  return nil
}
```

Reply:
Will correct this